tests/console/gpg.pm

```diff
@@ -32,33 +41,91 @@ sub run {
     wait_still_screen 1;
     type_string "1\n";
     wait_still_screen 1;
-    type_string "2048\n";
+    type_string "$key_size\n";
     wait_still_screen 1;
     type_string "0\n";
     wait_still_screen 1;
     type_string "y\n";
     wait_still_screen 1;
-    type_string "Test User\n";
+    type_string "$user_name\n";
     wait_still_screen 1;
-    type_string "user\@example.com\n", 1;
+    type_string "$email\n", 1;
     type_string "\n";
     wait_still_screen 1;
     save_screenshot;
     type_string "O\n";
 
     assert_screen("gpg-enter-passphrase");
+
     # enter wrong passphrase
     type_string "REALSECRETPHRASE\n";
     assert_screen("gpg-incorrect-passphrase");
     type_string "\t\n", 1;
+
     # enter correct passphrase
-    type_string "R34LS3CR3TPHR4S3\n";
+    type_string "$passwd\n";
     wait_still_screen 1;
-    type_string "R34LS3CR3TPHR4S3\n";
+    type_string "$passwd\n";
     wait_still_screen 1;
-    # list gpg keys
-    validate_script_output("gpg --list-keys", sub { m/\[ultimate\] Test User <user\@example\.com>/ });
+    if (get_var("FIPS") || get_var("FIPS_TS") && $key_size == 1024) {
+        assert_screen("key-generation-failed");
+    }
+    else {
+        # list gpg keys
+        validate_script_output("gpg --list-keys", sub { m/\[ultimate\] $user_name \<$email\>/ });
+    }
+}
+
+sub gpg_encrypt_file {
+
+    my ($key_size) = @_;
+    my $user_name  = $username . " " . $key_size;
+    my $user_id    = $user_name . " " . "\<$email\>";
+
+    # prepare a text file
+    assert_script_run("touch gtest.txt");
+
+    # Should generate encrypted file gtest.txt.gpg
+    assert_script_run("gpg2 -r \"$user_id\" -e gtest.txt");
+
+    # check encrypted file gtest.txt.gpg
+    assert_script_run("ls | grep gtest.txt.gpg");
+
+    # Decrypt function can be work
+    script_run("gpg2 -u \"$user_id\" -d gtest.txt.gpg", 0);
+    wait_still_screen 1;
+    type_string "$passwd\n";
+    wait_still_screen 1;
+
+    # Should generate signature file gtest.txt.asc
+    script_run("gpg2 -u \"$user_id\" --clearsign gtest.txt", 0);
+    wait_still_screen 1;
+    type_string "$passwd\n";
+    wait_still_screen 1;
+    assert_script_run("ls | grep gtest.txt.asc");
+
+    # Verify the signature
+    assert_script_run("gpg2 -u \"$user_id\" --verify gtest.txt.asc");
+
+    # clean up
+    assert_script_run("ls | grep gtest | xargs -i rm -f {}");
+}
+
+sub run {
+    select_console 'user-console';
+
+    # gpg key generated and file encrypted with two size of key
+    gpg_generate_key(2048);
+    gpg_encrypt_file(2048);
+    gpg_generate_key(3072);
+    gpg_encrypt_file(3072);
```

Comment:
From your failed verified run, I see that implicitly use $_ may cause mistake, so directly assign a  the $key_size variable in loop :)